middleware.go

```diff
@@ -0,0 +1,100 @@
+package jwt
+
+import (
+	"github.com/Digital-Insight-Technologies-Ltd/gin-jwt-cognito/models"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-errors/errors"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+)
+
+// UserClaimsKey is the key for user claims in our context
+const UserClaimsKey = "userClaims"
+
+// CognitoClaimsMiddleware is a middleware for Gin that extracts user claims from
+// AWS Cognito.
+func CognitoClaimsMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		tok, ok := c.GetQuery("Authorization")
+
+		if !ok {
+			c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": "Missing Authorization header"})
+			return
+		}
+
+		claims, err := parseClaims(tok[7:])
```

Comment:
Add a check to ensure the token string is at least 7 characters long before slicing with `tok[7:]` to prevent potential runtime panics. For example:
  if len(tok) < 7 {
      c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": "Invalid Authorization header"})
      return
  }